### DISCUSSION TITLE
CCButton - smart pointers and redesignating buttons

### DISCUSSION THREAD

**Discussion by nigjel** [tttapa/Control-Surface#742] [Q&A]
I am making a foot pedal with 4 footswitches. By default they are all latching which works fine (many thanks!!) Each footswitch can be redesignated as a momentary switch independently of the other 3. Following on from article #129 I have used smart pointers and the mechanism seems to work fine except that only the 'on' control is sent. The relevant lines of code are :

```
fs1 = make_unique<CCButton>(8, MIDI_CC::General_Purpose_Controller_1);    // replace default with momentary instance & initialise
fs1->begin();
```

Control_Surface.begin(), Control_Surface.connectDefaultMIDI_Interface(),  and  Control_Surface.loop(),  have all been declared as you might expect from the fact that CCButtonLatched works faultlessly. 

I'm guessing that the problem is with the parameters as I have found no way to add the channel as I would like to do. Do I need to modify CCButton.hpp?? I'm sure I've simply missed something obvious so hope someone can put me straight! My apologies in advance for any errors in formatting.

**Comment by tttapa** [comment]
You can use
```cpp
make_unique<CCButton>(8, MIDIAddress {MIDI_CC::General_Purpose_Controller_1, CHANNEL_2});
```
Or
```cpp
MIDIAddress addr {MIDI_CC::General_Purpose_Controller_1, CHANNEL_2};
make_unique<CCButton>(8, addr);
```

**Comment by nigjel** [comment]
many thanks!!

---
www.nigelfreake.eu
